ui/src/containers/Header/index.tsx

```diff
@@ -31,13 +31,55 @@ import { AUTH_BASE_URL, REDIRECT_URI } from '../../config/constants';
 import { IProvider } from '../../store/provider';
 import { titleCase } from '../../common/titlecase';
 import AlertDisplay from '../../components/AlertDisplay';
+import { AuthCodeProps, IError } from '../../store/auth';
 
 const Header: React.FC = observer(() => {
   const { user, providers } = useMst();
   const history = useHistory();
   const [isModalOpen, setIsModalOpen] = React.useState(false);
   const [disable, setDisable] = React.useState(false);
 
+  const detailPagePath = `detailPagePath`;
+
+  useEffect(() => {
```

Comment:
Any reason we are moving and handling all the code to header and not parser url

Reply:
these code are related to login so I had moved it to header as parse url component manage parsing of url params

Reply:
Since it is related to login, then shouldn't we move this to login component ?

Reply:
we don't have login component and handling login within header component